rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -420,8 +428,30 @@ public boolean createNodeSource(@HeaderParam("sessionid") String sessionId,
                 }
             }
         }
-        return rm.createNodeSource(nodeSourceName, infrastructureType, infraParams, policyType, policyParams)
-                 .getBooleanValue();
+        try {
+            nsState.setResult(rm.createNodeSource(nodeSourceName,
+                                                  infrastructureType,
+                                                  infraParams,
+                                                  policyType,
+                                                  policyParams)
+                                .getBooleanValue());
+        } catch (RuntimeException ex) {
+            nsState.setResult(false);
+            nsState.setErrorMessage(cleanDisplayedErrorMessage(ex.getMessage()));
+            nsState.setStackTrace(StringEscapeUtils.escapeJson(getStackTrace(ex)));
+
+        } finally {
+            return nsState;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this return statement from this finally block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1143)